### ISSUE TITLE
Adding harmony support for hardhat-etherscan

### ISSUE THREAD

**Issue by victaphu** [NomicFoundation/hardhat#2337]
Hi,

We've added support for hardhat-etherscan on harmony and created a small patch to support the harmony network (change in two files, chainconfig.ts and types.ts). I would like to push this as a PR as a lot of developers have been asking for hardhat-etherscan support to make contract verification much easier on the harmony network.

Below are some details on what has been done so far:
Documentation: [Verifying proxies using hardhat and harmony](https://victa-kwok-wai-phu.gitbook.io/harmony-contract-verification-service/verifying-proxy-contracts/transparent-proxies)
Fork: [hardhat with harmony](https://github.com/victaphu/hardhat/tree/master)

If this is acceptable I can use this issue to push a PR into your repository.

Thanks!

Victa

**Comment by alcuadrado** [comment]
Hey @victaphu, we accept PRs into hardhat-etherscan as long as they work with the existing architecture and follow [this guide](https://github.com/nomiclabs/hardhat/blob/master/packages/hardhat-etherscan/CONTRIBUTING.md).

**Comment by victaphu** [comment]
Great thank you I will follow that guide and update here when each step is completed.

For the testing, can you share which repo you're using for the tests?

**Comment by victaphu** [comment]
Hi @alcuadrado I have added the PR following the guide provided and sent 1 ONE token to the address in the guide.

This can be viewed on explorer at: https://explorer.harmony.one/tx/0xa364c416e3b36c09c3121e90ecf01a7ad989495be03500395fce1b8e6dc97c5b

**Comment by victaphu** [comment]
Hi @alcuadrado, @feuGeneA 
Just wanted to touch base on this to see if there were any progress. The PR is pushed and just waiting on review.

Best,
Victa

**Comment by victaphu** [comment]
Hi, closing this as the PR has been merged into the main repository
Thanks!
